### PR TITLE
network: check access mode of key file

### LINKED ISSUE #13516: systemd-networkd fails completely if just one of the config files is not accessible
Would it be possible to skip files that can't be opened due to EPERM without failing the entire systemd-networkd daemon? As it stands, the behavior of systemd-networkd is not very friendly.

First, systemd told me:

```
systemd-networkd[67775]: /etc/systemd/network/wg0.netdev has 0644 mode that is too permissive, please adjust the access mode.
```

I changed permission to 0600, and then:

```
Sep 10 04:56:46 l6 systemd[1]: Failed to start Network Service.
Sep 10 04:56:46 l6 systemd[1]: systemd-networkd.service: Failed with result 'exit-code'.
Sep 10 04:56:46 l6 systemd[1]: systemd-networkd.service: Main process exited, code=exited, status=1/FAILURE
Sep 10 04:56:46 l6 systemd-networkd[67765]: Could not load configuration files: Permission denied
```

See here, the fopen only skips config file on ENOENT error:

https://github.com/systemd/systemd/blob/f392c0656626824814704bdd624e02d56953dc2b/src/network/netdev/netdev.c#L663

A (requested) permission change on a non-essential interface should not cut all the network connectivity of the machine, completely.

### PR DESCRIPTION
**Pull request by yuwata**
Closes #13516.

### CODE REVIEW

File: src/network/networkd-network.c [hunk 1]
```diff
@@ -349,7 +349,7 @@ int network_load_one(Manager *manager, const char *filename) {
 
         network = new(Network, 1);
         if (!network)
-                return log_oom();
+                return -ENOMEM;
```
Comment (ssahani) [inline src/network/networkd-network.c]: Why drop all log_oom() rest looks good.

Comment (yuwata) [reply src/network/networkd-network.c]: Because, the function sometimes logs, and sometimes not. Anyway, I dropped the change.

**Comment by yuwata** [comment]
@sshanai Updated. PTAL.
- Also checks wireguard key file.
- revert dropping `log_oom()`.

**Comment by ssahani** [comment]
LGTM Thanks. Once ci passed we can merge

**Pull request merged**